### PR TITLE
fix(codebuild): add BatchPutCodeCoverages permission to Project by default

### PR DESCRIPTION
**Pull request by karthikns16**
----
Issues: Indicate issues fixed via: "Fixes https://github.com/aws/aws-cdk/issues/10750"
*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-codebuild/lib/report-group.ts [hunk 1]
```diff
@@ -46,6 +46,7 @@ abstract class ReportGroupBase extends cdk.Resource implements IReportGroup {
         'codebuild:CreateReport',
         'codebuild:UpdateReport',
         'codebuild:BatchPutTestCases',
+        'codebuild:BatchPutCodeCoverages',
```
Comment (skinny85) [inline packages/@aws-cdk/aws-codebuild/lib/report-group.ts:49]: This one I actually am less sure of... can you please revert this change?

**Review by skinny85 — DISMISSED** [review]
Thanks for the contribution @karthikns16 !

I have one thing I want to change here though.

File: packages/@aws-cdk/aws-codebuild/lib/report-group.ts (hunk 1 above)
Comment (karthikns16) [reply packages/@aws-cdk/aws-codebuild/lib/report-group.ts]: @skinny85 .Took care of it. Had some issue with my local builds had to rely on AWS Codebuild. Please check. 
I also think the report-group.ts should be updated to support codecoverage soon to maintain parity.

**Review by skinny85 — APPROVED** [review]
Thanks @karthikns16!

File: packages/@aws-cdk/aws-codebuild/lib/report-group.ts (hunk 1 above)
Comment (skinny85) [reply packages/@aws-cdk/aws-codebuild/lib/report-group.ts]: Perhaps, but that's a separate issue (predicated on the fact that currently [`ReportGroup` hard-codes the `Type` property to `'TEST'`](https://github.com/aws/aws-cdk/blob/cf268218163a6d25cc6bbe7375b9332f87062631/packages/%40aws-cdk/aws-codebuild/lib/report-group.ts#L128-L128)).

Comment (karthikns16) [reply packages/@aws-cdk/aws-codebuild/lib/report-group.ts]: Yeah, I am happy to contribute to that as well. Let me know if you would like to have a separate issue reported and tracked to closure?

Comment (skinny85) [reply packages/@aws-cdk/aws-codebuild/lib/report-group.ts]: Yes please 🙂
